src/Perl6/Grammar.nqp

```diff
@@ -4332,19 +4333,32 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
     token infix:sym<min>  { <sym> >> <O(|%tight_or_minmax)> }
     token infix:sym<max>  { <sym> >> <O(|%tight_or_minmax)> }
 
+    token ternary-explain {
+        || <?before '::' <.-[=]>> { self.typed_panic: "X::Syntax::ConditionalOperator::SecondPartInvalid", second-part => "::" }
```

Comment:
Why not? That's being used almost everywhere, and it's just a syntactic thing to make the stuff aligned.